src/components/Garden/Feed/WrapToken.js

```diff
@@ -39,74 +47,71 @@ function WrapToken({ onClaimRewards, onUnwrapToken, onWrapToken }) {
     onClaimRewards()
   }, [onClaimRewards, rewardsLink])
 
+  const carouselItems = [
+    <Token
+      balance={wrappableToken.accountBalance}
+      loading={loading}
+      mode="wrap"
+      onClick={onWrapToken}
+      token={wrappableToken.data}
+    />,
+    <Token
+      balance={token.accountBalance}
+      loading={loading}
+      mode="unwrap"
+      onClick={onUnwrapToken}
+      token={token.data}
+    />,
+    <Token
+      balance={earnedRewards}
+      loading={!earnedRewards}
+      mode="claim"
+      onClick={handleClaimRewards}
+      token={wrappableToken.data}
+    />,
+  ]
+
+  const handleItemSelected = useCallback(
+    index => {
+      if (index === carouselItems.length - 1) {
+      }
+    },
+    [carouselItems.length]
+  )
+
   return (
     <Box
       css={`
         ${!compactMode && `margin-bottom: ${3 * GU}px;`}
       `}
     >
-      <div
-        css={`
-          display: flex;
-          flex-direction: row;
-          justify-content: center;
-        `}
-      >
-        <Token
-          balance={wrappableToken.accountBalance}
-          loading={loading}
-          mode="wrap"
-          onClick={onWrapToken}
-          token={wrappableToken.data}
+      <div>
+        <Carousel
+          itemWidth={18 * GU}
+          itemHeight={tabletMode ? 24 * GU : 22 * GU}
+          itemSpacing={3 * GU}
+          items={carouselItems}
+          onItemSelected={handleItemSelected}
         />
-        <LineSeparator border={theme.border} />
-        <div>
-          <Token
-            balance={token.accountBalance}
-            loading={loading}
-            mode="unwrap"
-            onClick={onUnwrapToken}
-            token={token.data}
-          />
-          {earnedRewards.gt('0') && (
-            <div
-              css={`
-                margin-top: ${3 * GU}px;
-                padding-top: ${2 * GU}px;
-                border-top: 1px solid ${theme.border};
-              `}
-            >
-              Earned rewards:{' '}
-              <span
-                css={`
-                  color: ${theme.positive};
-                `}
-              >
-                {formatTokenAmount(earnedRewards, wrappableToken.data.decimals)}{' '}
-                {wrappableToken.data.symbol}
-              </span>{' '}
-              <Button
-                label="Claim"
-                mode="strong"
-                wide
-                onClick={handleClaimRewards}
-                css={`
-                  margin-top: ${1.5 * GU}px;
-                `}
-              />
-            </div>
-          )}
-        </div>
       </div>
     </Box>
   )
 }
 
 function Token({ balance, loading, mode, onClick, token }) {
+  const theme = useTheme()
+
   const wrapMode = mode === 'wrap'
-  const icon = wrapMode ? unwrappedIcon : wrappedIcon
+  const claimMode = mode === 'claim'
```

Comment:
Whenever we have 3 possible values we can create a mapping on top based on the mode. 

```
const modeAttributes = {
  wrap: { icon: unwrappedIcon, button: { mode: 'strong', label: 'Wrap' } },
  unwrap: {
    icon: wrappedIcon,
    button: { mode: 'strong', label: 'Unwrap' },
    hint:
      'This amount can be used to vote on proposals. It can be unwrapped at any time.',
  },
  claim: {
    button: { mode: 'normal', label: 'Claim' },
    icon: claimRewardsIcon,
  },
}
```

In the example went ahead and added also the hint so we can use it as the condition for rendering the Help

Reply:
I didn't understand how to use it when passing as a prop 😅